### DISCUSSION TITLE
Desafios Técnicos

### DISCUSSION THREAD

**Discussion by fredericksilva** [frontendbr/forum#472] [Discussão]
Então escrevi um desafio técnico, porém ainda não estou tendo muita adesão, gostaria de informações de vocês de como tenho que classificar o desafio.
 Por nivel técnico ou por necessidades reias da posição. 

##### Resumindo o desafio que escrevi é em React, com os seguintes requisitos:

- Consumir uma api existente
- montar um componente 
- exibir o estado da api no componente 
- test unitario do componente 
- test de api usando um mock 

Ao meu ver é um desafio de nivel médio, porém não sei se estou certo.
Me ajuda minha gente! ❤️

**Comment by marceloogeda** [comment]
@fredericksilva, não sei se eu entendi a sua pergunta. Você se refere a um desafio/test para preencher uma posição?

Se sim, na minha opinião o desafio deve ser feito para avaliar o nível técnico do candidato, você pode sim, utilizar de uma necessidade real da posição nesse desafio, porém, o que deverá ser avaliado é o conhecimento técnico, até mesmo porque se o mesmo precisar atuar em outro contexto ele consiga se desenvolver igual ou tão bem quanto.  Por isso talvez deva ser legal deixar o problema e ver/entender como o profissional chegou na solução. E com base na solução de cada problema, ter uma régua bem definida para poder avaliar o nível.

Espero não ter viajado muito na resposta hehe. 
Abraço!

**Comment by fredericksilva** [comment]
@marceloogeda Obrigado! Sim esse é o foco, fiz o desafio que ao meu entender qualquer dev caminhando de junior para pleno consegue fazer, tive feedback do time interno que o teste não é complicado e não tem pegadinhas. Só quero analisar o código e comportamento do desenvolvedor em relação ao comprometimento de boas praticas, boa comunicação. 

Como o teste não tem limite de tempo e a posição é uma que esta em constante processo de contratação e possível o cara que esta aplicando para vaga estude antes de fazer o teste ou estude fazendo e se ele comital podemos analisar todo o processo de evolução.

Isso que estou esperando dos candidatos, vamos ver se funciona. (Geralmente testes assusta mais do que impulsiona o dev na carreira)

**Comment by bernardodiasc** [comment]
sugestão: forneça as especificações do que o candidato vai ter que desenvolver. algo que aborde alguns desafios nivelados com a necessidade da vaga, mas que seja simples o suficiente pra ser feito em no maximo 3 dias, sendo que 3 dias nao sao full time, a disponibilidade das pessoas varia muito.

**Comment by felipefialho** [comment]
@fredericksilva Sabe uma boa?

Procura alguns repositórios com os testes para contratações em grandes empresas.

É uma forma divertida e bastante útil de se testar e pode usar como base para desenvolver seus proprios testes.

**Comment by fredericksilva** [comment]
Muito obrigado @LFeh e @bernardodiasc.
O teste que escrevi, não analisa apenas princípios técnicos. Também quero ver como o desenvolver lida com o code review... Tenho limitador no meu teste que pode acabar sendo um break point e que necessariamente o desenvolvedor tem que ter Github para poder participar. 

--- Também analiso o repo antes de enviar o teste, se o candidato não tem nada na linguagem do teste no Repo e mesmo assim quer fazer o desafio eu envio. Mas se o candidato tem um repo bem escrito e estruturado. Nem envio o teste (porque esse é o candidato ideal).

- Analiso as contribuições também.(as vezes acho que sou meio critico d+ rs)

**Comment by RatoX** [comment]
@fredericksilva se tu vai oferecer una vaga foda pra pessoa acho bem valido essa criticadade.
